src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -742,35 +743,163 @@ public override ClaimsPrincipal ValidateToken(string token, TokenValidationParam
             if (tokenParts.Length != JwtConstants.JwsSegmentCount && tokenParts.Length != JwtConstants.JweSegmentCount)
                 throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX12741, token)));
 
-            ClaimsPrincipal claimsPrincipal = null;
-            SecurityToken signatureValidatedToken = null;
-
+            ClaimsPrincipal claimsPrincipal;
+            SecurityToken signatureValidatedToken;
             if (tokenParts.Length == JwtConstants.JweSegmentCount)
             {
                 var jwtToken = ReadJwtToken(token);
                 var decryptedJwt = DecryptToken(jwtToken, validationParameters);
-                var innerToken = ValidateSignature(decryptedJwt, validationParameters);
-                jwtToken.InnerToken = innerToken;
-                signatureValidatedToken = jwtToken;
-                claimsPrincipal = ValidateTokenPayload(innerToken, validationParameters);
+                claimsPrincipal = ValidateToken(decryptedJwt, jwtToken, validationParameters, out signatureValidatedToken);
             }
             else
             {
-                signatureValidatedToken = ValidateSignature(token, validationParameters);
-                claimsPrincipal = ValidateTokenPayload(signatureValidatedToken as JwtSecurityToken, validationParameters);
+                claimsPrincipal = ValidateToken(token, null, validationParameters, out signatureValidatedToken);
             }
 
             validatedToken = signatureValidatedToken;
             return claimsPrincipal;
         }
 
+        /// <summary>
+        ///  Private method for token validation, responsible for:
+        ///  (1) Obtaining a configuration from the <see cref="TokenValidationParameters.ConfigurationManager"/>.
+        ///  (2) Revalidating using the Last Known Good Configuration (if present), and obtaining a refreshed configuration (if necessary) and revalidating using it.
+        /// </summary>
+        /// <param name="token">The JWS string, or the decrypted token if the token is a JWE.</param>
+        /// <param name="outerToken">If the token being validated is a JWE, this is the <see cref="JwtSecurityToken"/> that represents the outer token.
+        ///  If the token is a JWS, the value of this parameter is null.
```

Comment:
>null [](http://example.com/codeflow?start=67&length=4)

supernit: <see langword="null" />

Reply:
hmm the github page seems to strip the html I was saying you could do see langword = "null" here instead. very much a nit.